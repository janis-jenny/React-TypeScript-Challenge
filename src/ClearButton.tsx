import React from 'react';
import Button from '@mui/material/Button';
import Stack from '@mui/material/Stack';

const ClearButton: React.FC<{clear: () => void}> = ({clear}) => {
  return (
    <Stack direction="row" spacing={2}>
      <Button variant="contained" onClick={clear} className="mx-1">Clear</Button>
    </Stack>
  )
}

export default ClearButton;
