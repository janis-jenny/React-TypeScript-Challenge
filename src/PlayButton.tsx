import React from 'react';
import Button from '@mui/material/Button';
import Stack from '@mui/material/Stack';

const PlayButton: React.FC<{toggleBtn: () => void, paused: boolean}> = ({toggleBtn, paused}) => {
  return (
    <Stack direction="row" spacing={2}>
      <Button variant="contained" onClick={toggleBtn} className="mx-1">{paused? "Play" : "Stop"}</Button>
    </Stack>
  )
}

export default PlayButton;