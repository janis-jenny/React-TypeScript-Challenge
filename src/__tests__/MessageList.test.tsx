import renderer from 'react-test-renderer';
import ClearButton from '../components/ClearButton';
import PlayButton from '../components/ClearButton';


describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <>
        <ClearButton clear={function (): void {
          throw new Error('Function not implemented.');
        } } />
      </>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <>
        <PlayButton clear={function (): void {
          throw new Error('Function not implemented.');
        } } />
      </>,
    );
    expect(tree).toMatchSnapshot();
  });
});
