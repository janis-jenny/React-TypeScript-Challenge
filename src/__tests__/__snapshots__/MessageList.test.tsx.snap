// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="css-e53awj-MuiStack-root"
>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root mx-1 css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Clear
  </button>
</div>
`;

exports[`App renders correctly 2`] = `
<div
  className="css-e53awj-MuiStack-root"
>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root mx-1 css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Clear
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
